Firestore/core/src/firebase/firestore/remote/serializer.h

```diff
@@ -226,6 +226,29 @@ class Serializer {
 
   std::string EncodeQueryPath(const model::ResourcePath& path) const;
 
+  void ValidateDocumentKeyPath(nanopb::Reader* reader,
+                               const model::ResourcePath& resource_name) const;
+  model::DocumentKey DecodeKey(nanopb::Reader* reader,
+                               const model::ResourcePath& resource_name) const;
+
+  model::FieldValue::Map::value_type DecodeFieldsEntry(
```

Comment:
Alternatively, I could keep these as free or static functions and pass the database ID as an argument, but it seemed like doing a workaround for no clear reason.